File: aspn-py/src/aspn23/metadata_satnav_obs.py
"""
This code is generated via firehose.
DO NOT hand edit code.  Make any changes required using the firehose repo instead.
"""


from dataclasses import dataclass
from typing import Optional

from numpy.typing import NDArray

from .aspn_base import AspnBase
from .type_metadataheader import TypeMetadataheader
from .type_timestamp import TypeTimestamp


@dataclass
class MetadataSatnavObs(AspnBase):
    """
    Metadata for satnav_obs

    ### Attributes

    info - TypeMetadataheader:
            Standard ASPN metadata header.

    time_of_validity - TypeTimestamp:
            Time at which the measurement is considered to be valid.

    deltarange_interval_start - Optional[Optional[float]]:
            Start time of the interval over which the deltarange is measured with respect to the
            measurement time. This value must be provided if the deltarange measurement is provided.

    deltarange_interval_stop - Optional[Optional[float]]:
            Stop time of the interval over which the deltarange is measured with respect to the
            measurement time. This value must be provided if the deltarange measurement is provided.
            If the deltarange integration period is the same as the measurement time, this value
            should be 0.
    """

    info: TypeMetadataheader
    time_of_validity: TypeTimestamp
    deltarange_interval_start: Optional[float]
    deltarange_interval_stop: Optional[float]

File: aspn-c/src/aspn23/MetadataSatnavObs.c

/*
 * This code is generated via firehose.
 * DO NOT hand edit code.  Make any changes required using the firehose repo instead
 */

#include "MetadataSatnavObs.h"

Aspn23MetadataSatnavObs* ASPN_NULLABLE
aspn23_metadata_satnav_obs_new(Aspn23TypeMetadataheader* info,
                               Aspn23TypeTimestamp* time_of_validity,
                               float deltarange_interval_start,
                               float deltarange_interval_stop) {
	Aspn23MetadataSatnavObs* self =
	    (struct Aspn23MetadataSatnavObs*)malloc(sizeof(Aspn23MetadataSatnavObs));
	if (NULL == self) return NULL;

	Aspn23TypeMetadataheader* info_prep = aspn23_type_metadataheader_copy(info);
	self->info                          = *info_prep;
	free(info_prep);

	Aspn23TypeTimestamp* time_of_validity_prep = aspn23_type_timestamp_copy(time_of_validity);
	self->time_of_validity                     = *time_of_validity_prep;
	free(time_of_validity_prep);

	self->deltarange_interval_start = deltarange_interval_start;
	self->deltarange_interval_stop  = deltarange_interval_stop;
	return self;
}

Aspn23MetadataSatnavObs* ASPN_NULLABLE
aspn23_metadata_satnav_obs_copy(Aspn23MetadataSatnavObs* input) {
	Aspn23TypeMetadataheader* info_prep = aspn23_type_metadataheader_copy(&input->info);
	Aspn23TypeTimestamp* time_of_validity_prep =
	    aspn23_type_timestamp_copy(&input->time_of_validity);
	Aspn23MetadataSatnavObs* out = aspn23_metadata_satnav_obs_new(info_prep,
	                                                              time_of_validity_prep,
	                                                              input->deltarange_interval_start,
	                                                              input->deltarange_interval_stop);
	aspn23_type_metadataheader_free(info_prep);
	aspn23_type_timestamp_free(time_of_validity_prep);
	return out;
}

void aspn23_metadata_satnav_obs_free(void* pointer) {
	Aspn23MetadataSatnavObs* self = (Aspn23MetadataSatnavObs*)pointer;
	if (NULL == self) return;
	aspn23_metadata_satnav_obs_free_members(self);
	free(self);
}

void aspn23_metadata_satnav_obs_free_members(Aspn23MetadataSatnavObs* self) {
	if (NULL == self) return;
	aspn23_type_metadataheader_free_members(&self->info);
	aspn23_type_timestamp_free_members(&self->time_of_validity);
}
